apps/email/js/cards/cmp/autocomplete_source.js

```diff
@@ -0,0 +1,48 @@
+'use strict';
+define(function(require) {
+
+var contacts = navigator.mozContacts;
+
+function makeResult(query, contacts) {
+  return {
+    query: query,
+    contacts: contacts
+  };
+}
+
+/**
+ * Finds matches for the given query in the name and email for known people.
+ * Returns a promise that resolves to an object that has the query and a
+ * "match" array of mozContact entries. The match array will be a zero length
+ * array for no matches. See this bug about expanding the data source beyond
+ * contacts matches:
+ * https://bugzilla.mozilla.org/show_bug.cgi?id=1188710
+ * @param  {String} query the name/email query to use.
+ * @return {Promise}
+ */
+return function match(query) {
+  return new Promise(function(resolve, reject) {
+    if (!contacts || !query) {
+      return resolve(makeResult(query, []));
+    }
+
+    var request = contacts.find({
+      filterBy: ['name', 'email'],
```

Comment:
instead of 'name', I think we want to use 'givenName' and 'familyName'.  There are indices over these too, and "name" appears to be a composite created by the contacts app by taking the 0th entries of the givenName and familyName entries, trim()ing them, then smooshing them together with a space between them (and trimming again in case one is missing).

(So from our perspective, 'name' is really just a duplicate of 'givenName' if there is a givenName, or it's a duplicate of familyName if not.  The downside of favoring both of these is that if searching for "John Doe", if the user decides to type the whole name for some reason, once they get to "John D", 'name' would have kept matching, but familyName/givenName will refuse to give any results at that point.)